src/ng-select/lib/items-list.ts

```diff
@@ -393,9 +393,17 @@ export class ItemsList {
             });
             parent.children = children;
             parent.value = groupValue(key, children.map(x => x.value));
+            parent.disabled = parent.disabled || this._disabledChildren(parent);
```

Comment:
why the whole group should be disabled when at least 1 child is disabled?

Reply:
Also when you have `selectableGroup`, should it allow select it even if every child is disabled?

Reply:
It prevents you from selecting a disabled item. The UX is inconsistent if you can't select the item but it ends up getting selected when you click the group.

Reply:
I created a repo with this StackBlitz: https://angular-sdpzjc.stackblitz.io
The experience can be weird since you can't unselect the disabled item unless you re-select the entire group and then unselect the group.

Reply:
> 
> 
> Also when you have `selectableGroup`, should it allow select it even if every child is disabled?

This is a tough one. When you have selectableGroupAsModel=false I think disabling the group makes sense. If the group itself is selectable then perhaps it would make sense to select the group?

Reply:
In my project, our disabled items are ones that we never want the user to be able to modify directly. That's why it's a problem when those items get selected when a user clicks the group. But I could see why you would want that feature.

Perhaps this should be configuration dependent?

Reply:
I think this should be handled differently. If an item is disabled, click on a group shouldn't select it. It shouldn't prevent to select other items and group itself 🤔

Reply:
Well, you can still select the other items (manually), but I see your point.

When would we consider the group to be selected? If all non-disabled items are selected? In my opinion I would only consider the group selected if all items are selected.

Would clicking on the group select all items except for the disabled items?

Reply:
`selectableGroupAsModel=false` with this setup, I think we should allow selecting the group and just skip disabled items. This way group should not be marked as selected, but it will allow selecting all nondisabled children at once.

`selectableGroup` this one is a bit trickier, but since with disabled child option we can't have our whole group selected, technically it is the same as `selectableGroupAsModel=false`. Meaning - group selection will result in all non-disabled items selected and group itself won't be marked as selected.

Makes sense?

Reply:
Sounds good! I'll put that together.